### ISSUE TITLE
Replace travis with github-actions

### ISSUE THREAD

**Issue by ssbarnea** [pytest-dev/pytest-html#322]
I created this in because I want to exemplify why I think in 2020, it makes more sense to use Github Action than Travis.

There is nothing preventing us from running both in parallel for a longer period of time, and even ignoring the results from one of them if we want.

I can easily raise a PTs I already did this for lots of repos and I know the tricks.

## Update 2020-10

Changed scope from trying... to switching...

**Comment by BeyondEvil** [comment]
> I created this in because I want to exemplify why I think in 2020, it makes more sense to use Github Action than Travis.
> 
> There is nothing preventing us from running both in parallel for a longer period of time, and even ignoring the results from one of them if we want.
> 
> I can easily raise a PTs I already did this for lots of repos and I know the tricks.

I would really appreciate a PR with some Github Actions! 😊 

It would be awesome if you could comment it extensively to expedite the learning curve for me (and future contributors). Especially any tricks. 👍

**Comment by gnikonorov** [comment]
+1 to @BeyondEvil's response

**Comment by ssbarnea** [comment]
I think we all seen how nice github actions works. Maybe is time to effectively drop Travis and switch the release to github actions. Less complexity can only help the project.

@BeyondEvil @davehunt What do you think?

I cannot do it myself because I do not have access to project settings or pypi. One needs to add project-specific-tokens for both pypi instances that look like:

![](https://sbarnea.com/ss/Screen-Shot-2020-10-17-13-35-46.98.png)

Ideally we should use a bot account for that, but I see none configured. You could reuse the `pytest-dev` bot for that.

**Comment by gnikonorov** [comment]
Regarding the above, I think we should drop travis and move to github actions

**Comment by BeyondEvil** [comment]
I'm happy to make the switch over to only using GHA. I also think using the pytest-dev is a good idea as long as project owners have the access they need to it.

With that said, @davehunt has the final say.

**Comment by davehunt** [comment]
> With that said, @davehunt has the final say.

Don't let me get in the way of the using the best available tools. If you're all happy, I'm certainly not going to stop you.

**Comment by BeyondEvil** [comment]
Where can I see the pytest-dev in use with GHA? @ssbarnea

**Comment by BeyondEvil** [comment]
I need your pypi username @ssbarnea

What's the username of the pytest-dev bot @nicoddemus ? Will I have access to it? Do I need access to it (to make it the release bot)?

We should use [this](https://github.com/marketplace/actions/pypi-publish) and [this](https://pypi.org/help/#apitoken)

**Comment by nicoddemus** [comment]
The bot right now doesn't have a user name on PyPI yet, there's been some discussion in https://github.com/pytest-dev/meta/issues/4 but that has stalled.

What I usually do is create a scoped token in my own account and use that; anyone with release rights on PyPI can create and replace that token if needed.

**Comment by BeyondEvil** [comment]
Ah, got it. Thanks @nicoddemus  🙏

**Comment by gnikonorov** [comment]
It looks like uploading to test pypi fails, @ssbarnea: 

```
/usr/bin/docker run --name e5c35918e63663af24e6ab6512533d077c8e0_b82f1c --label 1e5c35 --workdir /github/workspace --rm -e PY_COLORS -e TOXENV -e pythonLocation -e LD_LIBRARY_PATH -e INPUT_PASSWORD -e INPUT_REPOSITORY_URL -e INPUT_USER -e INPUT_PACKAGES_DIR -e INPUT_VERIFY_METADATA -e INPUT_SKIP_EXISTING -e INPUT_VERBOSE -e HOME -e GITHUB_JOB -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_REPOSITORY_OWNER -e GITHUB_RUN_ID -e GITHUB_RUN_NUMBER -e GITHUB_RETENTION_DAYS -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_SERVER_URL -e GITHUB_API_URL -e GITHUB_GRAPHQL_URL -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e GITHUB_PATH -e GITHUB_ENV -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -e ACTIONS_RUNTIME_URL -e ACTIONS_RUNTIME_TOKEN -e ACTIONS_CACHE_URL -e GITHUB_ACTIONS=true -e CI=true -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/_temp/_runner_file_commands":"/github/file_commands" -v "/home/runner/work/pytest-html/pytest-html":"/github/workspace" 1e5c35:918e63663af24e6ab6512533d077c8e0  "__token__" "***" "https://test.pypi.org/legacy/" "dist" "true" "false" "false"
Checking dist/pytest_html-2.1.2.dev46+gca5ddbc-py3-none-any.whl: PASSED, with warnings
  warning: `long_description_content_type` missing. defaulting to `text/x-rst`.
Checking dist/pytest-html-2.1.2.dev46+gca5ddbc.tar.gz: PASSED, with warnings
  warning: `long_description_content_type` missing. defaulting to `text/x-rst`.
Uploading distributions to https://test.pypi.org/legacy/
Uploading pytest_html-2.1.2.dev46+gca5ddbc-py3-none-any.whl

  0%|          | 0.00/31.3k [00:00<?, ?B/s]
100%|██████████| 31.3k/31.3k [00:00<00:00, 99.4kB/s]
NOTE: Try --verbose to see response content.
HTTPError: 400 Bad Request from https://test.pypi.org/legacy/
'2.1.2.dev46+gca5ddbc' is an invalid value for Version. Error: Can't use PEP 440 local versions. See https://packaging.python.org/specifications/core-metadata for more information.

**Comment by gnikonorov** [comment]
Maybe we can change the package name to be a [pre release](https://www.python.org/dev/peps/pep-0440/#pre-releases) or [devlopmental release](https://www.python.org/dev/peps/pep-0440/#developmental-releases) identifier?

**Comment by ssbarnea** [comment]
I know what to do, I will do it tomorrow morning.

**Comment by BeyondEvil** [comment]
Is this fixed by #364 @ssbarnea ?
